lib/clientsidescripts.js

```diff
@@ -47,24 +47,46 @@ function wrapWithHelpers(fun) {
  *     be passed as a parameter.
  */
 functions.waitForAngular = function(rootSelector, callback) {
```

Comment:
We can't use `testForAngular` unless we modify the client side scripts strategy (as you suggested separately), so I'm punting on that one for now.

I split it into two different client side scripts, since it doesn't make sense to have it as an if/else case.